en/appendices/4-0-upgrade-guide.rst

```diff
@@ -89,13 +89,17 @@ composer commands:
     php composer.phar require --dev --update-with-dependencies "phpunit/phpunit:^8.0"
     php composer.phar require --update-with-dependencies "cakephp/cakephp:4.0.*"
 
-Application.php
+Others
 ===============
 
-Next, ensure your ``src/Application.php`` has been updated to have the same
-method signatures as the one found in cakephp/app. You can find the current
-`Application.php
-<https://github.com/cakephp/app/blob/4.x/src/Application.php>`__ on GitHub.
+Next, ensure your files has been updated to have the same
+method signatures as those found in cakephp/app. 
+
+#. ``src/Application.php`` at `Application.php <https://github.com/cakephp/app/blob/4.x/src/Application.php>`__ on GitHub.
```

Comment:
Please use links to the [4.0.4 tag](https://github.com/cakephp/app/tree/4.0.4), not 4.x

4.x is current CakePHP 4.5 which should not be used as a base to upgrade Cake 3 to.

Reply:
You could also just reference the whole app template as it is done in the [5.0 upgrade guide](https://book.cakephp.org/5/en/appendices/5-0-upgrade-guide.html#update-app-files-based-upon-latest-app-template) instead of manually listing all the files

Reply:
Should the links for [4.0.0 tag](https://github.com/cakephp/app/tree/4.0.4/config) be used, or 4.0.4 ?
Since it is only a few files that needs to be updated, I thought referencing them directly would be faster for users to upgrade.

Reply:
I agree that using concrete files will make them outdated and then lying over a short amount of time.
Better to stay more agnostic here.

Reply:
ping @yosus 
I would use the approach as in the upgrade guide.
Easiest to also maintain.